pyrmq/tests/test_consumer.py

```diff
@@ -116,3 +116,44 @@ def error_callback(*args):
 
             wait_for_result(response, [1])
             consumer.close()
+
+
+def should_get_message_with_higher_priority(priority_session: Publisher):
+    expected = {"test": "test0"}
+    for i in range(0, 10):
+        body = {"test": f"test{i}"}
+        priority_session.publish(body, priority=i)
+    response = {}
+
+    def callback(data):
+        if not response:
+            """
+            Checks the first message if it has the highest priority.
+            Dependent on ``x-max-priority`` argument. Test max priority is 5
+            Messages with priority 5 and above will be treated as priority 5.
+            Thus, they will be queued after the first message received with the highest priority
+            Example 1:
+                message1 = {"test": "test", "priority": 10}
+                message2 = {"test": "test", "priority": 5}
+                message3 = {"test": "test", "priority": 8}
+            message1 will be on the queue first, followed by message2 then message3
+            Example 2:
+                message1 = {"test": "test", "priority": 4}
+                message2 = {"test": "test", "priority": 8}
+                message3 = {"test": "test", "priority": 5}
+            message2 will be on the queue first, followed by message3 then message1
+            """
+            first_expected = {"test": "test5"}
```

Comment:
As much as possible, we should not have hard-coded values in assertions.

We should determine the expected first data based on the max priority. Maybe when we're building test data, we compare priority to the max priority and the first that matches will be the expected first data.